src/query/plan/rewrite/index_lookup.hpp

```diff
@@ -522,6 +531,19 @@ class IndexLookupRewriter final : public HierarchicalLogicalOperatorVisitor {
 
   std::optional<LabelIx> FindBestLabelIndex(const std::unordered_set<LabelIx> &labels) {
     MG_ASSERT(!labels.empty(), "Trying to find the best label without any labels.");
+
+    for (const auto &[index_type, label, _] : index_hints_) {
+      if (index_type != IndexHint::IndexType::LABEL) continue;
+
+      if (!db_->LabelIndexExists(GetLabel(label))) {
+        throw QueryRuntimeException("Index for label {} doesn't exist", label.name);
```

Comment:
to aggresive, I would rather log it with spdlog::debug and proceed next to see what happens with the plan

Reply:
Fixed (same as above)